pkg/oc/cli/tag/tag.go

```diff
@@ -529,17 +523,23 @@ func (o TagOptions) Run() error {
 			}
 
 			if target.Spec.Tags == nil {
-				target.Spec.Tags = make(map[string]imageapi.TagReference)
+				target.Spec.Tags = []imagev1.TagReference{}
 			}
 
-			if oldTargetTag, exists := target.Spec.Tags[destTag]; exists {
+			if oldTargetTag, exists := imageutil.SpecHasTag(target, destTag); exists {
 				if oldTargetTag.Generation == nil {
 					// for servers that do not support tag generations, we need to force re-import to fetch its metadata
 					delete(target.Annotations, imageapi.DockerImageRepositoryCheckAnnotation)
 					istag.Tag.Generation = nil
 				}
 			}
-			target.Spec.Tags[destTag] = *istag.Tag
+			// update tag
+			for i := range target.Spec.Tags {
+				t := target.Spec.Tags[i]
+				if t.Name == destTag {
+					t = *istag.Tag
```

Comment:
we could probably break after we've found the tag to update

Reply:
Good point.